pkg/config/config.go

```diff
@@ -95,13 +94,96 @@ func (c *ConfigTx) Updated() *cb.Config {
 	return c.updated
 }
 
+// ComputeUpdate computes the ConfigUpdate from a base and modified config transaction.
+func (c *ConfigTx) ComputeUpdate(channelID string) (*cb.ConfigUpdate, error) {
+	if channelID == "" {
+		return nil, errors.New("channel ID is required")
+	}
+
+	updt, err := computeConfigUpdate(c.base, c.updated)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compute update: %v", err)
+	}
+
+	updt.ChannelId = channelID
+
+	return updt, nil
+}
+
+// ChannelConfiguration returns a channel configuration value from a config transaction.
+func (c *ConfigTx) ChannelConfiguration() (Channel, error) {
+	channelGroup := c.base.ChannelGroup
+	var err error
+
+	_, ok := channelGroup.Values[ConsortiumKey]
+	var consortium string
+	if ok {
+		consortiumProto := &cb.Consortium{}
+		err := unmarshalConfigValueAtKey(channelGroup, ConsortiumKey, consortiumProto)
+		if err != nil {
+			return Channel{}, err
+		}
+		consortium = consortiumProto.Name
+	}
+
+	_, ok = channelGroup.Groups[ApplicationGroupKey]
+	var application Application
+	if ok {
+		application, err = c.GetApplicationConfiguration()
```

Comment:
You may want to consider if we even need the if ok check above for these. Considering you're not erroring out here if the ApplicationGroup does not exist but just moving on with an empty Application, should we instead just call GetApplicationConfiguration and defer to its error if the group doesn't exist?

Reply:
The intend here is to support retrieve both `application channel` and `system channel`, therefore, it did not error out in the first place, but just returned empty Application and move on. We can definitely split this up into two functions, which we can target specific channel types. Any recommendations?

Reply:
Makes sense, hadn't considered system channel yet.